### PR TITLE
Add list functionality to mlflow models.

### PR DESCRIPTION
**Pull request by mberk06**
## Changes

* Add `ModelListResponse` and `ModelListRequest` structs
* Add a List function to the model resource
* Add 3 tests

## Tests

- [x] `make test` run locally
- [ ] relevant change in `docs/` folder
- [ ] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [ ] using Go SDK

### CODE REVIEW

**Comment by mberk06** [comment]
@alexott I'm starting with a list function for the mlflow exporter functionality - Rohit suggested this is a good first step. I've started working on other functionality as well. Thanks!

File: mlflow/resource_mlflow_model.go [hunk 1]
```diff
@@ -50,6 +61,45 @@ func NewModelsAPI(ctx context.Context, m any) ModelsAPI {
 	return ModelsAPI{m.(*common.DatabricksClient), ctx}
 }
 
+func (a ModelsAPI) ListByFilter(filter string, order_by []string, page_token string) ([]Model, error) {
```
Comment (nfx) [inline mlflow/resource_mlflow_model.go:64]: please use Go SDK, and not implement this manually.

probably a good idea to migrate those mlflow resources to SDK first.

**Review by nfx — CHANGES_REQUESTED** [review]

File: mlflow/resource_mlflow_model.go (hunk 1 above)
Comment (nfx) [reply mlflow/resource_mlflow_model.go:64]: https://pkg.go.dev/github.com/databricks/databricks-sdk-go@v0.5.0/service/mlflow#RegisteredModelsAPI.ListAll

**Comment by mberk06** [comment]
Closing and moving to SDK functionality.

**Pull request closed without merging**